lib/types/local-time.js

```diff
@@ -27,257 +29,209 @@ const nanoSecInMillis = Long.fromNumber(1000000);
+        let millis =
+            parseInt(parts[0], 10) * 3600000 + parseInt(parts[1], 10) * 60000;
+        let nanos;
+        if (parts.length === 3) {
+            const secParts = parts[2].split(".");
+            millis += parseInt(secParts[0], 10) * 1000;
+            if (secParts.length === 2) {
+                nanos = secParts[1];
+                //add zeros at the end
+                nanos = nanos + utils.stringRepeat("0", 9 - nanos.length);
+            }
+        }
+        return LocalTime.fromMilliseconds(millis, parseInt(nanos, 10) || 0);
+    }
 
-/**
- * Returns the string representation of the instance in the form of hh:MM:ss.ns
- * @returns {String}
- */
-LocalTime.prototype.toString = function () {
-    return formatTime(this._getParts());
-};
+    /**
+     * Uses the current local time (in milliseconds) and the nanoseconds to create a new instance of LocalTime
+     * @param {Number} [nanoseconds] A Number from 0 to 999,999, representing the time nanosecond portion.
+     * @returns {LocalTime}
+     */
+    static now(nanoseconds) {
+        return LocalTime.fromDate(new Date(), nanoseconds);
+    }
 
-/**
- * Gets the string representation of the instance in the form: hh:MM:ss.ns
- * @returns {String}
- */
-LocalTime.prototype.toJSON = function () {
-    return this.toString();
-};
+    /**
+     * Uses the provided local time (in milliseconds) and the nanoseconds to create a new instance of LocalTime
+     * @param {Date} date Local date portion to extract the time passed since midnight.
+     * @param {Number} [nanoseconds] A Number from 0 to 999,999, representing the nanosecond time portion.
+     * @returns {LocalTime}
+     */
+    static fromDate(date, nanoseconds) {
+        if (!util.types.isDate(date)) {
+            throw new Error("Not a valid date");
+        }
+        //Use the local representation, only the milliseconds portion
+        const millis =
+            (date.getTime() + date.getTimezoneOffset() * -60000) % millisInDay;
+        return LocalTime.fromMilliseconds(millis, nanoseconds);
+    }
 
-/**
- * @returns {Array.<Number>}
- * @ignore
- */
-LocalTime.prototype._getParts = function () {
-    if (!this._partsCache) {
-        //hours, minutes, seconds and nanos
-        const parts = [0, 0, 0, 0];
-        const secs = this.value.div(nanoSecInSec);
-        //faster modulo
-        //total nanos
-        parts[3] = this.value.subtract(secs.multiply(nanoSecInSec)).toNumber();
-        //seconds
-        parts[2] = secs.toNumber();
-        if (parts[2] >= 60) {
-            //minutes
-            parts[1] = Math.floor(parts[2] / 60);
-            parts[2] = parts[2] % 60;
+    /**
+     * Uses the provided local time (in milliseconds) and the nanoseconds to create a new instance of LocalTime
+     * @param {Number} milliseconds A Number from 0 to 86,399,999.
+     * @param {Number} [nanoseconds] A Number from 0 to 999,999, representing the time nanosecond portion.
```

Comment:
This appears to be optional parameter. Maybe make the documentation more informative about that?

Reply:
`@param {Number} [nanoseconds]` name of variable in brackets means optional parameter. I think that is enough?